app/models/apple/episode_delivery_status.rb

```diff
@@ -1,5 +1,21 @@
 module Apple
   class EpisodeDeliveryStatus < ApplicationRecord
     belongs_to :episode, class_name: "::Episode"
+
+    def self.update_status(episode, attrs)
+      new_status = (episode.apple_episode_delivery_status&.dup || new(episode: episode))
+      new_status.assign_attributes(attrs)
+      new_status.save!
+      episode.apple_episode_delivery_statuses.reset
+      new_status
+    end
+
+    def increment_asset_wait
+      self.class.update_status(episode, asset_processing_attempts: (asset_processing_attempts || 0) + 1)
```

Comment:
Ah, so the "asset wait" is the number of processing attempts we've made on the episode thus far.  Or ... the latest version of this episode's media, i suppose.

Reply:
The media version increments are tracked in another column `source_media_version_id`. e.g. if you group on the episode id you can track the media versions that the episode have gone through:

```
SELECT count(source_media_version_id) count, episode_id
FROM (
        SELECT DISTINCT source_media_version_id,
                        episode_id
        FROM apple_episode_delivery_statuses
        WHERE source_media_version_id IS NOT NULL
) q
GROUP BY episode_id
HAVING count(source_media_version_id) > 1
ORDER BY count DESC
```